kotlin-analysis-core/src/main/kotlin/org/jetbrains/research/ml/kotlinAnalysis/psi/PsiProvider.kt

```diff
@@ -15,12 +19,16 @@ import org.jetbrains.research.ml.kotlinAnalysis.util.isKotlinRelatedFile
 object PsiProvider {
 
     fun extractMethodsFromProject(project: Project): List<KtNamedFunction> {
-        // We should reverse the list with method since we handle all methods separately
+        // We should reverse the list with method since we handle all elements separately
         // and we can invalidate the previous one
-        return extractPsiFiles(project).map { collectPsiMethods(it) }.flatten().reversed()
+        return extractElementsOfTypeFromProject(project, KtNamedFunction::class.java).reversed()
     }
 
-    private fun extractPsiFiles(project: Project): MutableSet<PsiFile> {
+    fun extractImportDirectiveFromProject(project: Project): List<KtImportDirective> {
```

Comment:
Why you created this function? Maybe can we call `extractElementsOfTypeFromProject(project, KtImportDirective::class.java)` directly?